### ISSUE TITLE
allowSyntheticDefaultImports doesn't work when mixing commonJS and ES6 modules

### ISSUE THREAD

**Issue by jerry2013** [microsoft/TypeScript#36856]
I've been trying to use typescript to add type checking to a fairly large js application. Given its history it has a mix of commonJS and ES6 style modules.

I enabled `allowSyntheticDefaultImports` in jsconfig.json but I'm flooded with TS1192 errors. I created a simple test and found that the error would pop up if importing a commonJS module which uses another commonJS module.

**TypeScript Version:**  Version 3.9.0-dev.20200215

**Search Terms:** allowSyntheticDefaultImports TS1192

**Code**

```ts
// app.js
import MyTest from "./MyTest"; // this gives TS1192 error
import MyMod from "./MyMod";
import Mod from "./Mod";
MyTest.test();
MyMod.test();
Mod.test();

// MyTest.js
import MyMod from 'MyMod';
module.exports = {
  test: () => MyMod.test()
};

// MyMod.js
module.exports = {
  test: () => console.log('test')
};

// Mod.js
import MyMod from 'MyMod';
export default {
  test: () => MyMod.test()
}
```

**Expected behavior:**
`allowSyntheticDefaultImports` should allow nested commonJS modules and does not emit TS1192 error.

**Actual behavior:**
```
./node_modules/.bin/tsc --pretty -p ./jsconfig.json 
app.js:1:8 - error TS1192: Module '"/facetalk/test/MyTest"' has no default export.

1 import MyTest from "./MyTest";
         ~~~~~~
```

**Comment by RyanCavanaugh** [comment]
@sandersn seems like a bug to me but please correct me if I'm wrong

**Comment by sandersn** [comment]
Typescript doesn't support mixing commonjs and ES modules in a single file, so you should rewrite `MyTest` like so:

```js
const MyMod = require('./MyMod')
module.exports = {
  test: () => MyMod.test()
}
```

After I did that your example worked fine, although it's likely to crash at runtime unless you switch to `esModuleInterop: true`, because that emits a compatibility helper, unlike `allowSyntheticDefaultImports`, which just quashes the error.
